content/en/docs/concepts/technical concepts.md

```diff
@@ -18,8 +34,47 @@ Beamstack acts as a low-code abstraction layer that allow users to deploy ML pip
 
 ## Monitoring and Visualization
 
-Beamstack provides a monitoring stack consisting of Grafana and Prometheus that allows users to visualize the state of containers and resources within their pipelines.
+Beamstack includes an integrated monitoring stack featuring Grafana and Prometheus, designed to provide comprehensive visibility into the state of containers and resources within ML pipelines. This monitoring stack enables users to track performance, troubleshoot issues, and optimize resource utilization effectively.  
+
+This entails:  
+
+<details>
+  <summary><b>Prometheus Integration</b></summary>
+  <ul>
+    <li><b>Real-Time Metrics Collection</b>: Prometheus is employed for real-time collection of metrics from various components within the ML pipeline environment. It scrapes data from containers, services, and infrastructure to provide detailed insights into resource usage, performance, and operational status.</li>
+  </ul>
+  <ul>
+    <li><b>Efficient Data Storage</b>: Prometheus uses a time-series database to store metric data, ensuring efficient querying and retrieval of historical performance data. This facilitates trend analysis and long-term monitoring.</li>
+  </ul>
+</details>
+
+<details>
+  <summary><b>Grafana Visualization</b></summary>
+  <ul>
+    <li><b>Custom Dashboards</b>: Grafana integrates with Prometheus to offer customizable dashboards where users can visualize collected metrics through various types of charts, graphs, and tables. Users can create and configure dashboards to display key performance indicators (KPIs) and other critical metrics relevant to their ML pipelines.</li>
+  </ul>
+  <ul>
+    <li><b>Interactive Data Exploration</b>: Grafana provides interactive data exploration capabilities, allowing users to drill down into specific metrics, filter data, and generate ad-hoc queries. This interactive functionality aids in detailed analysis and troubleshooting.</li>
+  </ul>
+</details>
+
+<details>
+  <summary><b>Container and Resource Monitoring</b></summary>
```

Comment:
will be good to add the screen shot

Reply:
<img width="1728" alt="monitoring" src="https://github.com/user-attachments/assets/a50f4ba9-aba9-4d7f-bdb7-48a4477e0385">
@kennedyuche You can use this, feel free to re-size.